script.js

```diff
@@ -1,4 +1,371 @@
+  for (let i = 0; i <= totalPlayers; i++) {
+    outputMessage += `<br>${String(playersHands[i].playerNum).padEnd(
+      10,
+      "\xa0"
+    )}\xa0${playersHands[i].score}\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0${
+      playersHands[i].wallet
+    }\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 ${printHand(i)} `;
+  }
+  return outputMessage;
+};
+
+//To determine the next player based on whether their turn is done
+var determineNextPlayer = function () {
+  for (let i = 1; i <= totalPlayers; i++) {
+    if (playersHands[i].turnDoneFlag == 0) {
+      return i;
+    }
+  }
+};
+
+//Print out the hand of the player
+var printHand = function (playerIndex) {
+  let outputMessage = "";
+  for (let i = 0; i < playersHands[playerIndex].hand.length; i++) {
+    outputMessage += `${playersHands[playerIndex].hand[i].name}${playersHands[playerIndex].hand[i].suit}\xa0\xa0`;
+  }
+  return outputMessage;
+};
+
+//Calculate the score of the player based on their index with 0 being the dealer and 1 being player 1 and so on
+var calcScore = function (playerIndex) {
+  let noOfAce = 0;
+  playersHands[playerIndex].score = 0;
+  for (let i = 0; i < playersHands[playerIndex].hand.length; i++) {
+    score = playersHands[playerIndex].hand[i].rank;
+    if (score > 10) {
+      score = 10;
+    }
+    if (score == 1) {
+      noOfAce += 1;
+    }
+    playersHands[playerIndex].score += score;
+  }
+  for (let i = 0; i < noOfAce; i++) {
+    if (playersHands[playerIndex].score <= 11) {
+      playersHands[playerIndex].score += 10;
+    }
+  }
+};
+
+//Run the dealer's turn
+var dealerTurn = function () {
+  while (playersHands[0].turnDoneFlag == 0) {
+    if (playersHands[0].score < 16) {
+      playersHands[0].hand.push(shuffledDeck.pop());
+      calcScore(0);
+    } else {
+      playersHands[0].turnDoneFlag = 1;
+    }
+  }
+};
+
+//Settle who wins and the wager
+var tabulateScore = function () {
+  for (let i = 1; i <= totalPlayers; i++) {
+    if (playersHands[i].blackjackFlag == 0) {
+      if (playersHands[0].score <= 21 && playersHands[i].score > 21) {
+        playersHands[i].wallet -= playersHands[i].bet;
+      } else if (playersHands[0].score <= 21) {
+        if (playersHands[i].score - playersHands[0].score > 0) {
+          playersHands[i].wallet += playersHands[i].bet;
+        }
+        if (playersHands[i].score - playersHands[0].score < 0) {
+          playersHands[i].wallet -= playersHands[i].bet;
+        }
+      } else if (playersHands[0].score > 21) {
+        if (playersHands[i].score <= 21) {
+          playersHands[i].wallet += playersHands[i].bet;
+        }
+      }
+    }
+  }
+};
+
+//To initialise the number of players
+var playerInit = function (input) {
+  totalPlayers = input;
+  for (let i = 0; i <= totalPlayers; i++) {
+    let playerIndex = i;
+    if (i == 0) {
+      playerIndex = "Dealer";
+    }
+    var player = {
+      playerNum: playerIndex,
+      wallet: 100,
+      hand: [],
+      turnDoneFlag: 0,
+      turn: 1,
```

Comment:
May I know what is the turn used for? Imo I think you do not need it because I can't find anywhere that you had to use it.

Reply:
We need to not show the next player's hands to the current player so this is a state so that when we change players, they have to press submit first and it will show them their hand of two cards